### PR TITLE
Fix cmocka unit test for linux

### LINKED ISSUE #12242: Fix unit tests for linux and server
|EPIC|Component|
|---|---|
| https://github.com/wazuh/wazuh/issues/9103 | FIM |

After changes done in the epic, we needed fix c existent tests .

The purpose of this issue is to look at all the problems in the C unit tests that have been broken with this change for linux and server

### PR DESCRIPTION
**Pull request by FrancoRivero**
|EPIC|Component|
|---|---|
| https://github.com/wazuh/wazuh/issues/9103 | FIM |

### Description

Fix cmocka unit tests for linux environment

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
  - [x] MAC OS X
- [x] Unit tests passed
- [x] Review logs syntax and correct language

- Memory tests for Linux
  - [x] Valgrind (memcheck and descriptor leaks check)
  - [x] cppcheck (static code analysis)
  - [x] Coverage (unit tests coverage code)
  - [x] Sformat (code style)
  - [x] Scan build
 

Closes #12242

### CODE REVIEW

File: src/unit_tests/syscheckd/test_syscheck.c [hunk 1]
```diff
@@ -228,18 +258,19 @@ void test_Start_win32_Syscheck_syscheck_disabled_2(void **state) {
...
     expect_string(__wrap__minfo, formatted_msg, FIM_FILE_SIZE_LIMIT_DISABLED);
 
     expect_string(__wrap__minfo, formatted_msg, FIM_DISK_QUOTA_LIMIT_DISABLED);
 
     snprintf(info_msg, OS_MAXSTR, "Started (pid: %d).", getpid());
     expect_string(__wrap__minfo, formatted_msg, info_msg);
-    expect_assert_failure(Start_win32_Syscheck());
+   expect_function_call(__wrap_os_wait);
+    expect_function_call(__wrap_start_daemon);
+    assert_int_equal(Start_win32_Syscheck(), 0);
```
[the remark this answers is not in the document]
Comment (FrancoRivero) [reply src/unit_tests/syscheckd/test_syscheck.c:273]: Done

File: src/unit_tests/syscheckd/test_syscheck.c [hunk 2]
```diff
@@ -300,16 +331,16 @@ void test_Start_win32_Syscheck_dirs_and_registry(void **state) {
 
     expect_string(__wrap__minfo, formatted_msg, "(6004): No diff for file: 'Diff'");
 
-    expect_value(__wrap_fim_db_init, memory, 0);
-    will_return(__wrap_fim_db_init, NULL);
+    expect_wrapper_fim_db_init(0, 300, 100000, 100000, true);
 
-    expect_string(__wrap__merror_exit, formatted_msg, "(6698): Creating Data Structure: sqlite3 db. Exiting.");
+    //expect_string(__wrap__merror_exit, formatted_msg, "(6698): Creating Data Structure: sqlite3 db. Exiting.");
```
[the remark this answers is not in the document]
Comment (FrancoRivero) [reply src/unit_tests/syscheckd/test_syscheck.c]: Done

**Review by antoniomanuelfr — APPROVED** [review]

**Review by JoseAntonioMHerrera — APPROVED** [review]
LGTM!

**Review by jotacarma90 — APPROVED** [review]
LGTM

File: src/unit_tests/wazuh_modules/github/CMakeLists.txt [hunk 3]
```diff
@@ -14,7 +14,11 @@ list(APPEND tests_flags "-Wl,--wrap,_mwarn -Wl,--wrap,_mdebug1 -Wl,--wrap,_merro
                         -Wl,--wrap,_mtwarn -Wl,--wrap,_mtdebug1 -Wl,--wrap,_mtdebug2 -Wl,--wrap,StartMQ -Wl,--wrap,sleep \
                         -Wl,--wrap,OS_IsValidIP -Wl,--wrap,OSMatch_Execute -Wl,--wrap,wm_sendmsg -Wl,--wrap,OSRegex_Compile \
                         -Wl,--wrap,wm_state_io -Wl,--wrap,OSRegex_Execute -Wl,--wrap,OSRegex_Execute_ex -Wl,--wrap,OSMatch_Compile \
-                        -Wl,--wrap,OSRegex_FreePattern -Wl,--wrap,wurl_http_request -Wl,--wrap,strftime -Wl,--wrap,localtime_r")
+                        -Wl,--wrap,OSRegex_FreePattern -Wl,--wrap,wurl_http_request -Wl,--wrap,strftime -Wl,--wrap,localtime_r \
+                        -Wl,--wrap,fim_sync_push_msg -Wl,--wrap,fim_run_integrity -Wl,--wrap,fim_db_remove_path \
```
Comment (chemamartinez) [inline src/unit_tests/wazuh_modules/github/CMakeLists.txt:18]: Is it needed to include the FIM wrappers for the Github tests?

File: src/syscheckd/include/syscheck.h [hunk 4]
```diff
@@ -960,21 +960,4 @@ int fim_generate_delete_event(const char *file_path,
...
- * @brief Send a control synchronization message
- * @param component Name of the component.
- * @param msg Type of the message.
- * @param id Sync session counter (timetamp).
- * @param start First key in the list.
- * @param top Last key in the list.
- * @param tail Key of the first key in the next sublist.
- * @param checksum Checksum of this list.
- */
-void fim_send_sync_control(const char *component,
```
[the remark this answers is not in the document]
Comment (FrancoRivero) [reply src/syscheckd/include/syscheck.h:973]: Done ! This function was deleted in every place

File: src/syscheckd/CMakeLists.txt [hunk 5]
```diff
@@ -96,6 +96,7 @@ if (UNIX)
 endif(UNIX)
 
 if(UNIT_TEST)
+add_definitions(-DWAZUH_UNIT_TESTING)
```
[the remark this answers is not in the document]
Comment (FrancoRivero) [reply src/syscheckd/CMakeLists.txt]: Done

File: src/unit_tests/wazuh_modules/github/CMakeLists.txt (hunk 3 above)
Comment (FrancoRivero) [reply src/unit_tests/wazuh_modules/github/CMakeLists.txt]: No, this was cleared in the last commit

**Review by chemamartinez — APPROVED** [review]

**Pull request merged**